src/components/modals/CustomModal.tsx

```diff
@@ -0,0 +1,51 @@
+import { Modal } from "antd";
+import { Imodal } from "../../types/types";
+import StepOneRegistration from "./StepOneRegistration";
+import StepTwoRegistration from "./StepTwoRegistration";
+
+const CustomModal: React.FC<Imodal> = ({
+  isOpen,
+  onClose,
+  data,
+  setIsActiveModal,
+  label,
+  setPublicKey,
+  setSecretKey,
+  publicKey,
+  secretKey,
+}) => {
+  return (
+    <Modal
+      className={`modal-container-${label}`}
+      centered
+      open={isOpen}
+      onCancel={onClose}
+      footer={null}
+    >
+      {label === "generateKeyPair" && (
+        <StepOneRegistration
+          onClose={onClose}
+          setIsActiveModal={setIsActiveModal}
+          data={data}
+          label={label}
+          setSecretKey={setSecretKey}
+          setPublicKey={setPublicKey}
+        />
+      )}
+      {label === "confirmWallet" && (
+        <StepTwoRegistration
+          onClose={onClose}
+          setIsActiveModal={setIsActiveModal}
+          data={data}
+          label={label}
+          secretKey={secretKey}
+          publicKey={publicKey}
+          setSecretKey={setSecretKey}
+          setPublicKey={setPublicKey}
+        />
+      )}
```

Comment:
Can we make this more dynamic? What happens if tomorrow we've to add 4 more steps? This will increase and become bigger and bigger.

Reply:
Fixed in commit d5dd696371cd44e45af55bcbabd6811af380cde2.